crates/core/src/client/mod.rs

```diff
@@ -368,6 +370,29 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
         Ok(storage_value)
     }
 
+    /// Returns the amount which the sender is allowed to withdraw from the owner.
+    async fn token_allowance(
+        &self,
+        contract_address: Address,
+        account_address: Address,
+        spender_address: Address,
+    ) -> Result<TokenAllowance, EthApiError<P::Error>> {
+        let block_id = BlockId::Number(BlockNumberOrTag::Latest);
+
+        let contract_addr: Felt252Wrapper = contract_address.into();
+        let contract = EthereumErc20::new(contract_addr.into(), &self.kakarot_contract);
+
+        let handle = FutureTokenAllowance::<P, _>::new(contract.allowance(
+            account_address.into(),
+            spender_address.into(),
+            block_id,
+        ));
+
+        let allowance = handle.await;
+
+        Ok(allowance)
```

Comment:
This is a very strange syntax, why not just:
```suggestion
        contract.allowance(
            account_address.into(),
            spender_address.into(),
            block_id,
        ).await
```